shapeshift/web
# feat: time estimate UI for trade quotes

## Description

Updates quote UI to display estimated execution time where possible.

Updates the difference percentage to be the difference between the receive amount ignoring fees.

Note that the presence of a time estimate depends on market conditions and the amount - bigger amounts get time estimates more often.

Also note that the time estimate will be the same for both thorchain quotes except for extremely large amounts (~100eth)

## Pull Request Type


## Issue (if applicable)

closes #5109

## Risk

* Moderate risk of broken quotes generally
* Low risk of broken trades

## Testing

* Check estimated time is roughly correct
* Check trades and quotes still work

### Engineering

☝️

### Operations

☝️

## Screenshots (if applicable)

Case showing different time estimates for streaming and non-streaming thor swaps
![image](https://github.com/shapeshift/web/assets/125113430/0aa572eb-1d97-4264-a785-a70655ac0fa2)

Case showing mix of quotes with and without a time estimate (note: percentage has been fixed since this screenshot)
![image](https://github.com/shapeshift/web/assets/125113430/a4b95011-1740-4c86-9d29-329ea2b7f965)

Example of lifi time estimate appearing
![image](https://github.com/shapeshift/web/assets/125113430/1fa45aed-949e-4808-90d0-5d8a8a4b184f)
